source/base/mpi.cc

```diff
@@ -511,9 +511,82 @@ namespace Utilities
       const MPI_Comm &                 mpi_comm,
       const std::vector<unsigned int> &destinations)
     {
-      return compute_point_to_point_communication_pattern(mpi_comm,
-                                                          destinations)
-        .size();
+      auto destinations_unique = destinations;
+      std::sort(destinations_unique.begin(), destinations_unique.end());
+      destinations_unique.erase(std::unique(destinations_unique.begin(),
+                                            destinations_unique.end()),
+                                destinations_unique.end());
+
+      if (Utilities::MPI::min<unsigned int>(destinations.size() ==
+                                              destinations_unique.size(),
+                                            mpi_comm))
+        {
+          return compute_point_to_point_communication_pattern(mpi_comm,
```

Comment:
what does the documentation of that function say about duplicates? I tend to think that the two functions should have the same semantics regarding duplicates, given that their names are almost the same.

Reply:
Good point. The documentation states:

> destinations	The list of processors the current process wants to send information to. This list need not be sorted in any way. If it contains duplicate entries that means that multiple messages are intended for a given destination.

> A list of processors that have indicated that they want to send something to the current processor. The resulting list is not sorted. It may contain duplicate entries if processors enter the same destination more than once in their destinations list.

But, the function has the same problem as `compute_n_point_to_point_communications()`. I have fixed it.